### PR TITLE
Rust discord bot (workshop bounty)

### PR DESCRIPTION
**Pull request by anirudhb**
Adds a workshop about writing a Discord bot to run polls in Rust.

Workshop bounty guidelines:
- [x] You must be a club leader or an active community member in order to submit a workshop
 Active Hack Club Slack member
- [x] Your workshop must produce a compelling, specific, real project, using real code. It should not be a tutorial for a language or service. It should also not be a series of small projects with no clear ending or final product.
 Final product is a Discord bot allowing you to run polls.
- [x] Your workshop must be able to be completed in around 20 minutes, but it should be very easy for people to make big changes to it.
 Serenity docs can help people make changes easily. Workshop is completable in ~20-25min excluding further in-depth explanations.
- [x] At the end of your workshop, you must include links to at least 3 demos clearly showing the initial project being changed in big ways. These demos should serve as examples for how hackable your workshop is. Each demo should include a link to the project and a link to the source code.
- [x] Your workshop must involve code, and not require desktop software or hardware, since not all clubs will have those resources.
 Workshop is done on Repl.it.
- [x] At the beginning of your workshop, you must include the link to the final demo and the final code.
- [x] Your workshop must be in English at native-level proficiency. Phrasing and grammar must be perfect.
- [x] By default, all workshops should be written in Markdown or another format that can be easily be added to the Workshops page.
 Workshop written in Markdown
- [x] Each step in your workshop should be very clearly explained and should continually give context for where the user is in the workshop.
 Each step provides context for where to put code.
- [x] Your workshop must not be the same or too similar to any existing workshop. Make sure to check on this before you start writing your workshop—if your workshop is too similar to an existing one, it won't be able to be merged at all!
 Confirmed on Slack
- [x] Every piece of media (image, GIF, video, etc.) you include in your workshop should include alt text that clearly describes what's in the piece of media

Link to mock meeting: [here](https://hackclub.slack.com/archives/C01A9V5LPD0/p1606249208040200).

### CODE REVIEW

**Review by MatthewStanciu — CHANGES_REQUESTED** [review]
Hey @anirudhb—thanks so much for submitting this. This is a fantastic workshop and I'm really excited to finally have some Rust content on the Workshops page.

A few things I noticed:

* This workshop definitely doesn't take 20 minutes to complete. This is by far the longest workshop bounty ever submitted, and once merged it'll be one of the longest workshops on the workshops page. The length of this workshop is in line with others that have taken an hour to an hour and 20 minutes to complete, so I expect this workshop to take at least an hour to complete, (at least within a club setting).
  * Technically that means this workshop violates one of the rules of the workshop bounty. But I've decided to give an exception for this workshop because 1) it's a fantastic and super well-written workshop, 2) I think it's important to have Rust content on the website, and 3) I don't see any way you could cut this workshop down.
  * That said, having short, bite-sized, accessible Rust workshops is just as important to me as having workshops like this. So if you submit a workshop in the future, I highly recommend trying to write something that's shorter and more accessible
* This workshop is not a beginner workshop. I wouldn't recommend it to anyone just starting to learn Rust, and personally I got lost a few times while reading it. There are a lot of concepts that are a little glossed over—and for good reason, this workshop is very heavy, and explaining in-depth every little simple concept would make it way too long—but regardless, I think it's important to add a disclaimer listing the basic concepts that people should be familiar with, just to set expectations.
  * If you have time before the end of the year, I would _love_ to have a workshop that serves as a super effective intro to Rust :)

Can you make the few changes I requested and ping me on Slack once you've made the changes? Once you've made them I'll merge.

File: workshops/discord_bot_with_rust/README.md
Comment (anirudhb) [reply workshops/discord_bot_with_rust/README.md:1175]: This code block is explained in the description below. It's used to close the scope opened above, which relinquishes read access to the global data map (so we can take write access later, this is explained as well.)

**Comment by MatthewStanciu** [comment]
This is awesome!

**Pull request merged**
